### DISCUSSION TITLE
Unexpected render for section 2.6 (quad)

### DISCUSSION THREAD

**Discussion by PoneyUHC** [RayTracing/raytracing.github.io#1632] [Debugging]
Hello, 

I am currently working on the section 2.6 (adding quad primitives)
I see that this section is pretty new, so I was wondering if there was a bug of some sort.
The render I produce from the code (which I checked to be the same 4 times) is the following :
![image](https://github.com/user-attachments/assets/c161fc39-0a24-4ce2-9e32-56da23616a8d)

It just feels like the scene in not properly setup, orelse it would be a camera issue that I did not spot earlier but this really seems unlikely...

[Here is the code](https://github.com/PoneyUHC/PathTracer/commit/00588ef7344feeb17c1bcba6025e15d9f08e1467#) I am testing against (branch develop, commit 00588ef)

Would you have any idea of what is happening ? I'm a bit lost, as the new code amount is pretty limited.

Thanks !

**Comment by hollasch** [comment]
Our latest progression branch for book 2 is `progression/2/2024-0405`. We've confirmed that compiling and running after listing 54 (A new scene with quads) does indeed produce the render in the book.

To compare against the actual code state at that point, I'd recommend checking out commit d7c98ae and comparing against your code. Let us know if you find anything.

**Reply by PoneyUHC** [reply]
Hey, thanks for your reply.
I solved the issue, the cause was (once again) between the keyboard and the screen.

In the constructor I wrote `w` as a computation over `normal` instead of `n`, so all the dimensions were messed up.

Faulty : 
```cpp
normal = unit_vector(cross(u, v));
D = dot(normal, Q);
w = normal/ dot(normal, normal);
```

Right :
```cpp
auto n = cross(u, v);
normal = unit_vector(n);
D = dot(normal, Q);
w = n / dot(n,n);
```

My bad, and thanks again for pointing me to the code (which solved it almost first line), I can continue the journey :)

**Reply by hollasch** [reply]
Yeah, I remember that code being a bit tricky. Glad you figured it out.
